v1/runtime/runtime_test.go

```diff
@@ -1695,3 +1697,63 @@ func TestExtraDiscoveryOpts(t *testing.T) {
 		t.Error("expected extra discovery option to be called")
 	}
 }
+
+type h struct {
+	c topdown_cache.InterQueryCache
+}
+
+func (j *h) OnInterQueryCache(_ context.Context, c topdown_cache.InterQueryCache) error {
+	j.c = c
+	return nil
+}
+
+type i struct {
+	c topdown_cache.InterQueryValueCache
+}
+
+func (j *i) OnInterQueryValueCache(_ context.Context, c topdown_cache.InterQueryValueCache) error {
+	j.c = c
+	return nil
+}
+
+func TestCacheHooksOnServer(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), 2*time.Millisecond)
+	defer cancel() // NOTE(sr): The timeout will have been reached by the time `done` is closed.
+	testLogger := testLog.New()
+
+	h1 := h{}
+	h2 := i{}
+
+	params := NewParams()
+	params.Logger = testLogger
+	params.Addrs = &[]string{"localhost:0"}
+	params.Hooks = hooks.New(&h1, &h2)
+
+	rt, err := NewRuntime(ctx, params)
+	if err != nil {
+		t.Fatalf("Unexpected error %v", err)
+	}
+	initChannel := rt.Manager.ServerInitializedChannel()
+	done := make(chan struct{})
+	go func() {
+		rt.StartServer(ctx)
+		close(done)
+	}()
+	<-done
+	select {
+	case <-initChannel:
+		return
+	default:
+		t.Fatal("expected ServerInitializedChannel to be closed")
+	}
+	if h1.c == nil {
+		t.Errorf("expected non-nil inter-query cache")
+	}
+	if h2.c == nil {
+		t.Errorf("expected non-nil inter-query value cache")
+	}
+
+	for _, e := range testLogger.Entries() {
+		t.Log(e.Message)
+	}
```

Comment:
[question]: Just to double-check, this is all intended for error-case output, right? The `return` in the `select` block above here looks like it should result in early-exiting the test on success cases.